content/docs/user-guide/experiment-management/sharing-experiments.md

```diff
@@ -116,6 +116,25 @@ the flags to select a different set of experiments to push.
 
 </details>
 
+<details>
+
+### Push experiments automatically
+
+To push the experiment automatically at the end of a `dvc exp run` or
+`dvc exp save` set the configuration option `exp.auto_push` to `true`:
+
+```cli
+$ dvc config exp.auto_push true
+```
+
+or use the [environment variable](/doc/user-guide/env) `DVC_EXP_AUTO_PUSH`.
+
+By default, the experiments will be pushed to the remote `origin`. To change the
```

Comment:
```suggestion
By default, the experiments will be pushed to the remote `origin`. To change the
```

@skshetry You mentioned wanting to look for the upstream remote rather than use `origin`. Do you think it's worth changing the behavior or should we move forward with this as is?

Reply:
Looking at `dvc-studio-client` implementation, although it tries to figure out branch's upstream remote and fallback to "origin", it does not work with `dvc exp run` as it runs in detached HEAD. So it's always falling back to `origin`.

Ideally, I'd prefer the order to be something like follows:

1. `DVC_EXP_GIT_REMOTE`
2. `DVC_STUDIO_REPO_URL`
3. `STUDIO_REPO_URL`
4. `exp.git_remote`
5. `studio.repo_url`
6. `<upstream remote>`
7. `origin`

But it seems like it's not going to be trivial to get the baseline branch of the experiment currently running to figure out the upstream url, as it's running in a detached HEAD.

Although we could support `STUDIO_REPO_URL`, I think it's very inconvenient to expect user to pass a URL, instead of a simple `remote` name.

So, this effectively means that the order will be `DVC_EXP_GIT_REMOTE` -> `exp.git_remote` -> `origin`, which is what we have right now.

Reply:
All of these configs and env vars are a bit confusing. So, I think defaulting to `origin` might be a good default for now.

If a user needs to set up a custom remote, they have to setup two configs `exp.git_remote` and `studio.repo_url`.
(although we could add support for `exp.git_remote` on `dvclive`, `dvc-studio-client` and `dvc exp push`).

Reply:
We could also make `studio.repo_url` an alias for `exp.git_remote` and deprecate it, so you can specify either a URL or a git remote name.

Reply:
Does `exp.git_remote` work with a URL or only a remote name?